### PR TITLE
fixes(#5439): Build should enforce a required jdk version

### PR DESCRIPTION
**Pull request by tdiesler**
enforces JDK 17 or above

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by squakez — COMMENTED** [review]
The presence of the check is a good thing to have. However, we introduce a dependency on some additional tooling (`awk`) we may not have in all the different CI machines involved in the build. Let's see if the checks are green though.

**Comment by tdiesler** [comment]
Ok, lets try to do this in Go - its easier to maintain anyway

File: script/check_jdk_version.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		panic(fmt.Sprintf("Error: %v\n", err))
+	}
+
+	versionStr := string(output)
+	versionRegex := regexp.MustCompile(`version "?([1-9]+)(\.([0-9]+)){2,3}"?`)
+	matches := versionRegex.FindStringSubmatch(versionStr)
+	if len(matches) < 2 {
+		panic(fmt.Sprintf("Unable to determine Java version: %s\n", versionStr))
```
Comment (squakez) [inline script/check_jdk_version.go:22]: IMO we should not panic. We cannot test all the possible JDK out there to make sure they return the expected text.

File: script/check_jdk_version.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package main
```
Comment (squakez) [inline script/check_jdk_version.go:1]: We have a cmd/util directory where we should host any tool. Please, keep the same structure.

**Review by squakez — COMMENTED** [review]
-0 

I'm not a big fan of maintaining external tools. If we want to have it, let's make sure this is not panicking if something is not found as expected.

Also, if we want to be fair and do a proper tool, what Camel K really need for building is Maven, and Maven has to have a JAVA_HOME pointing to a JDK 17 based. So, If we want to do the things properly, we need to make sure to check the `java version` run on the JAVA_HOME bin and not the one in the path (which may be different).

File: script/check_jdk_version.go (hunk 2 above)
Comment (tdiesler) [reply script/check_jdk_version.go:1]: ok

File: script/check_jdk_version.go (hunk 1 above)
Comment (tdiesler) [reply script/check_jdk_version.go:22]: ok. How about we get the java version from maven i.e.

```
$ mvn -version
Apache Maven 3.9.6 (bc0240f3c744dd6b6ec2920b3cd08dcc295161ae)
Maven home: /opt/homebrew/Cellar/maven/3.9.6/libexec
Java version: 17.0.6, vendor: Azul Systems, Inc., runtime: /Library/Java/JavaVirtualMachines/zulu-17.jdk/Contents/Home
Default locale: en_GB, platform encoding: UTF-8
OS name: "mac os x", version: "14.4.1", arch: "aarch64", family: "mac"
```

Comment (squakez) [reply script/check_jdk_version.go:22]: Okey. Still make sure any parsing problem is not leading to a panic which would impede to complete a build.

**Review by squakez — APPROVED** [review]

**Pull request merged**